onnxruntime/core/providers/webnn/builders/model_builder.cc

```diff
@@ -272,6 +280,11 @@ Status ModelBuilder::AddOperandFromPersistMemoryBuffer(
   emscripten::val view = emscripten::val::undefined();
   emscripten::val desc = emscripten::val::object();
   switch (data_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_BOOL:
+      view = emscripten::val{emscripten::typed_memory_view(size / sizeof(uint8_t),
+                                                           reinterpret_cast<const uint8_t*>(dest))};
+      desc.set("type", emscripten::val("uint8"));
+      break;
     case ONNX_NAMESPACE::TensorProto_DataType_FLOAT16:
       view = emscripten::val{emscripten::typed_memory_view(size / sizeof(uint16_t),
                                                            reinterpret_cast<const uint16_t*>(dest))};
```

Comment:
Consider also adding int32/uint32 sooner than later, because it can be used in Slice and Gather (`Tind : tensor(int32), tensor(int64)`). It need not be in this CR, but I figured since we're updating this switch statement right now anyway?

https://github.com/onnx/onnx/blob/main/docs/Operators.md#inputs-3---5-1
https://github.com/onnx/onnx/blob/main/docs/Operators.md#outputs-53

Reply:
Yeah sure, I will add int32/uint32 later.